### PR TITLE
[1407] Course preview flag for 'mcb providers show'

### PR DESCRIPTION
**Pull request by benilovj**
### Context

`mcb` output for complex records (e.g. courses) can be hard to parse visually.

### Changes proposed in this pull request

Add an explicit renderer for courses, that can be reused across multiple `mcb` commands.

#### Without `-p` option

```sh
UCAS Preferences:
no preferences found

Provider Enrichments:
No data.

Provider Courses:
id | age_range | course_code | name          | profpost_flag             | program_type                   | qualification | start_date              | study_mode | accrediting_provider_id | provider_id | modular | english | maths | science | created_at              | updated_at              | changed_at
---|-----------|-------------|---------------|---------------------------|--------------------------------|---------------|-------------------------|------------|-------------------------|-------------|---------|---------|-------|---------|-------------------------|-------------------------|------------------------
10 |           | 825         | Mathematics   | postgraduate              | school_direct_training_prog... | pgce_with_qts | 2019-09-01 00:00:00     |            |                         | 18          | M       | 9       | 1     |         | 2019-02-23 23:45:24     | 2019-02-23 23:45:24     | 2019-04-11 20:15:08
11 |           | ECC         | Biology       | professional_postgraduate | higher_education_programme     | pgce_with_qts | 2019-09-01 00:00:00     | full_time  |                         | 18          |         | 9       | 3     |         | 2019-02-23 23:45:24     | 2019-02-23 23:45:24     | 2019-02-23 23:45:24
24 | primary   | 3XXD        | Primary (3-7) | postgraduate              | scitt_programme                | pgce_with_qts | 2019-09-01 00:00:00     | full_time  | 18                      | 18          |         | 3       | 3     |         | 2019-04-29 14:59:23     | 2019-04-29 14:59:23     | 2019-04-29 14:59:23
27 | primary   | 3XXE        | Primary (3-7) | postgraduate              | scitt_programme                | pgce_with_qts | 2019-09-01 00:00:00     | full_time  | 18                      | 18          |         | 3       | 3     |         | 2019-04-30 12:25:55     | 2019-04-30 12:25:55     | 2019-04-30 12:25:55
28 | primary   | CXXG        | Primaru       |                           |                                | pgce_with_qts | 2019-09-01 00:00:00     | full_time  |                         | 18          |         |         |       |         | 2019-04-30 16:42:24     | 2019-04-30 16:42:24     | 2019-04-30 16:42:24
```

#### With `-p` option

```sh
UCAS Preferences:
no preferences found

Provider Enrichments:
No data.

Provider Courses:
+--------------------+----------------------------------+
| Course code        | 825                              |
| Description        | PGCE with QTS                    |
| Provider           | A01                              |
| Accredited body    | Self-accrediting                 |
| Subjects           | Mathematics                      |
| Level              | secondary                        |
| Training locations | Wisozk, Predovic and Collier     |
| Start date         | Sep 2019                         |
| Route              | school_direct_training_programme |
| Age range          |                                  |
| Modular            | M                                |
| English            | 9                                |
| Maths              | 1                                |
| Science            |                                  |
+--------------------+----------------------------------+
+--------------------+------------------------------+
| Course code        | ECC                          |
| Description        | PGCE with QTS full time      |
| Provider           | A01                          |
| Accredited body    | Self-accrediting             |
| Subjects           | Biology                      |
| Level              | secondary                    |
| Training locations | Wisozk, Predovic and Collier |
| Start date         | Sep 2019                     |
| Route              | higher_education_programme   |
| Age range          |                              |
| Modular            |                              |
| English            | 9                            |
| Maths              | 3                            |
| Science            |                              |
+--------------------+------------------------------+
+--------------------+-------------------------+
| Course code        | 3XXD                    |
| Description        | PGCE with QTS full time |
| Provider           | A01                     |
| Accredited body    | A01                     |
| Subjects           |                         |
| Level              | secondary               |
| Training locations |                         |
...
```

### CODE REVIEW

File: lib/mcb/commands/providers/show.rb [hunk 1]
```diff
@@ -27,6 +27,6 @@
        'address2', 'address3', 'address4', 'postcode', 'telephone'
 
     puts "\nProvider Courses:"
-    tp provider.courses
+    provider.courses.map { |course| puts Terminal::Table.new rows: MCB::CourseShow.new(course).to_h }
```
Comment (misaka) [inline lib/mcb/commands/providers/show.rb]: Can we add a switch to make this optional? I like the compact version of this, and usually manage the wrapping by piping to a pager. Maybe we should invoke a pager automatically like git does.

Comment (benilovj) [reply lib/mcb/commands/providers/show.rb]: I've made the behaviour optional.

File: lib/mcb/commands/providers/show.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 name 'show'
 summary 'Show information about provider'
 param :code
+option :p, :'preview-courses', 'Show courses as a mini-preview of Find, instead of a database view.',
```
Comment (defong) [inline lib/mcb/commands/providers/show.rb]: just wondering do we wanna have it to generate the url for it as well?

**Comment by defong** [comment]
[New course creation through mcb](https://trello.com/c/QGQiBD7b/1407-new-course-creation-through-mcb)

**Comment by defong** [comment]
[New course creation through mcb](https://trello.com/c/QGQiBD7b/1407-new-course-creation-through-mcb)
